cmd_test/migrate_from_chroma_test.go

```diff
@@ -0,0 +1,165 @@
+	ctx := context.Background()
+
+	qdrantContainer := qdrantContainer(ctx, t, qdrantAPIKey)
+	defer func() {
+		if err := qdrantContainer.Terminate(ctx); err != nil {
+			t.Errorf("Failed to terminate Qdrant container: %v", err)
+		}
+	}()
+	chromaContainer := chromaContainer(ctx, t)
+	defer func() {
+		if err := chromaContainer.Terminate(ctx); err != nil {
+			t.Errorf("Failed to terminate Chroma container: %v", err)
+		}
+	}()
+
+	chromaHost, err := chromaContainer.Host(ctx)
+	require.NoError(t, err)
+	chromaPort, err := chromaContainer.MappedPort(ctx, chromaPort)
+	require.NoError(t, err)
+
+	qdrantHost, err := qdrantContainer.Host(ctx)
+	require.NoError(t, err)
+	qdrantPort, err := qdrantContainer.MappedPort(ctx, qdrantPort)
+	require.NoError(t, err)
+
+	chromaClient, err := chroma.NewHTTPClient(chroma.WithBaseURL("http://" + chromaHost + ":" + chromaPort.Port()))
+	require.NoError(t, err)
+
+	collection, err := chromaClient.GetOrCreateCollection(ctx, testCollectionName)
+	require.NoError(t, err)
+
+	testIDs := make([]chroma.DocumentID, totalEntries)
+	testEmbeddings := make([]embeddings.Embedding, totalEntries)
+	testDocuments := make([]string, totalEntries)
+	testMetadatas := make([]chroma.DocumentMetadata, totalEntries)
+
+	for i := 0; i < totalEntries; i++ {
+		testIDs[i] = chroma.DocumentID(fmt.Sprintf("%d", i+1))
+
+		randomVector := make([]float32, dimension)
+		for j := range randomVector {
+			randomVector[j] = rand.Float32()
+		}
+		testEmbeddings[i] = embeddings.NewEmbeddingFromFloat32(randomVector)
+
+		testDocuments[i] = fmt.Sprintf("test document %d", i+1)
+		meta, err := chroma.NewDocumentMetadataFromMap(map[string]interface{}{
+			sourceField: fmt.Sprintf("test%d", i+1),
+		})
+		require.NoError(t, err)
+		testMetadatas[i] = meta
+	}
+
+	err = collection.Add(
+		ctx,
+		chroma.WithTexts(testDocuments...),
+		chroma.WithIDs(testIDs...),
+		chroma.WithMetadatas(testMetadatas...),
+		chroma.WithEmbeddings(testEmbeddings...),
+	)
+	require.NoError(t, err)
+
+	qdrantClient, err := qdrant.NewClient(&qdrant.Config{
+		Host:   qdrantHost,
+		Port:   qdrantPort.Int(),
+		APIKey: qdrantAPIKey,
+	})
+	require.NoError(t, err)
+	defer qdrantClient.Close()
+
+	args := []string{
+		"chroma",
+		fmt.Sprintf("--chroma.url=http://%s:%s", chromaHost, chromaPort.Port()),
+		fmt.Sprintf("--chroma.collection=%s", testCollectionName),
+		fmt.Sprintf("--qdrant.url=http://%s:%s", qdrantHost, qdrantPort.Port()),
+		fmt.Sprintf("--qdrant.api-key=%s", qdrantAPIKey),
+		fmt.Sprintf("--qdrant.collection=%s", testCollectionName),
+		fmt.Sprintf("--migration.batch-size=%d", batchSize),
+		fmt.Sprintf("--migration.offsets-collection=%s", offsetsCollectionName),
+		fmt.Sprintf("--qdrant.id-field=%s", idField),
+		fmt.Sprintf("--qdrant.document-field=%s", documentField),
+		fmt.Sprintf("--qdrant.distance=%s", distance),
+		fmt.Sprintf("--qdrant.dense-vector=%s", denseVectorField),
+	}
+
+	command, err := cmd.NewParser(args)
```

Comment:
We discussed to invoke migration CLI as the user would do, but here again we are hooking up into the migration codebase and calling the tool as the library. I would expect that these kind of integration tests work as following:

* build migration image during the startup phase if not already built
* spin up dependencies
* run migration command with https://golang.testcontainers.org/features/creating_container/#run
* do needed assertions in the way how you are proposing in this PR

Reply:
But now that the inputs are the same as the user's. What better will invoking it from the shell do?

Reply:
Done in https://github.com/qdrant/migration/commit/62c80cec44adc2fe3798c5c71be6da8882493152.